src/extensions/membrane-orientation/behavior.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2018-2020 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Alexander Rose <alexander.rose@weirdbyte.de>
+ * @author Sebastian Bittrich <sebastian.bittrich@rcsb.org>
+ */
+
+import { ParamDefinition as PD } from '../../mol-util/param-definition';
+import { StructureRepresentationPresetProvider, PresetStructureRepresentations } from '../../mol-plugin-state/builder/structure/representation-preset';
+import { MembraneOrientationProvider } from '../../mol-model-props/computed/membrane-orientation';
+import { StateObjectRef } from '../../mol-state';
+import { Task } from '../../mol-task';
+import { PluginBehavior } from '../../mol-plugin/behavior';
+import { StructureSelectionQuery, StructureSelectionCategory } from '../../mol-plugin-state/helpers/structure-selection-query';
+import { MolScriptBuilder as MS } from '../../mol-script/language/builder';
+import { DefaultQueryRuntimeTable, QuerySymbolRuntime } from '../../mol-script/runtime/query/base';
+import { MembraneOrientationRepresentationProvider } from './representation';
+import { CustomPropSymbol } from '../../mol-script/language/symbol';
+import Type from '../../mol-script/language/type';
+import { isInMembranePlane } from '../../mol-model-props/computed/membrane-orientation/ANVIL';
+import { StructureProperties } from '../../mol-model/structure';
+import { Vec3 } from '../../mol-math/linear-algebra';
+import { HydrophobicityColorThemeProvider } from '../../mol-theme/color/hydrophobicity';
+
+export const MembraneOrientationData = PluginBehavior.create<{ autoAttach: boolean }>({
+    name: 'membrane-orientation-prop',
+    category: 'custom-props',
+    display: {
+        name: 'Membrane Orientation',
+        description: 'Initialize orientation of membrane layers. Data calculated with ANVIL algorithm.' // TODO add ' or obtained via RCSB PDB'
+    },
+    ctor: class extends PluginBehavior.Handler<{ autoAttach: boolean }> {
+        private provider = MembraneOrientationProvider
+
+        register(): void {
+            DefaultQueryRuntimeTable.addCustomProp(this.provider.descriptor);
```

Comment:
this (and also in unregister) is only needed when you actually have `this.provider.descriptor.symbols` defined (see rcsb/validation-report for example)

Reply:
How about moving the `isTransmembraneSymbol` to the file with the provider following the setup as rcsb/validation-report. Also, add the symbol to the custom-property Descriptor there.